analyses/fusion_filtering/03-Calc-zscore-annotate.R

```diff
@@ -144,55 +112,96 @@ ZscoredAnnotation<-function(standardFusionCalls=standardFusionCalls,zscoreFilter
+         # join the filtered expression values to the data frame keeping track of symbols
+         # for each sample-fusion name pair
+         dplyr::left_join(expression_sample_matched, by = c("Sample", "GeneSymbol")) %>%
+         # for each sample-fusion name pair, are all genes under the expression threshold?
+         dplyr::group_by(FusionName, Sample) %>%
+         dplyr::select(FusionName, Sample,zscore_value,gene_position) %>%
+         # cast to keep zscore and gene position
+         reshape2::dcast(FusionName+Sample ~gene_position,value.var = "zscore_value") %>%
+         # incase Gene2A/B dont exist like in STARfusion calls
+         add_column(!!!cols[setdiff(names(cols),names(.))]) %>%
+         # get annotation from z score
+         dplyr::mutate(note_expression_Gene1A = ifelse((Gene1A>zscoreFilter| Gene1A< -zscoreFilter),"differentially expressed","no change"),
+                       note_expression_Gene1B = ifelse((Gene1B>zscoreFilter| Gene1B< -zscoreFilter),"differentially expressed","no change"),
+                       note_expression_Gene2A = ifelse((Gene2A>zscoreFilter| Gene2A< -zscoreFilter),"differentially expressed","no change"),
+                       note_expression_Gene2B = ifelse((Gene2B>zscoreFilter| Gene2B< -zscoreFilter),"differentially expressed","no change")) %>%
+         dplyr::rename(zscore_Gene1A=Gene1A,
+                       zscore_Gene1B=Gene1B,
+                       zscore_Gene2A=Gene2A,
+                       zscore_Gene2B=Gene2B) %>%
+         # unique FusionName-Sample rows
+         # use this to filter the QC filtered fusion data frame
+         dplyr::inner_join(standardFusionCalls, by = c("FusionName", "Sample")) %>%
+         dplyr::distinct()
+       expression_annotated_fusions <- rbind(expression_annotated_fusions, expression_annotated_fusion_individual)
+     }
   return (expression_annotated_fusions)
 }
 
-# example run using GTEx
-# load GTEx data
-normData<-read_tsv(gtexMatrix) %>% rename(GeneSymbol=gene_symbol)
-
-# load standardaized fusion calls cohort
-standardFusionCalls<-readRDS(standardFusionCalls)
-
-# load expression Matrix for cohort
-expressionMatrix<-readRDS(expressionMatrix)
-expressionMatrix <- cbind(expressionMatrix, colsplit(expressionMatrix$gene_id, pattern = '_', names = c("EnsembleID","GeneSymbol")))
-
-# for cohort level run the expressionMatrix divided by broad_histology will be provided to the function as normData and expressionMatrix
+GTExZscoredAnnotation_filtered_fusions <- data.frame()
+for (j in (1:length(cohortInterest))) {
+  cohort_BSids <- read.delim(clinicalFile, header = TRUE, sep = "\t", stringsAsFactors = FALSE) %>%
+    # uses each value x in cohortInterest
+    filter(cohort == cohortInterest[j]) %>% filter(experimental_strategy == "RNA-Seq") %>%
+    filter(sample_type == "Tumor") %>% pull("Kids_First_Biospecimen_ID")
+  # use index to find the matched normal data
+  normData = normData[j]
```

Comment:
Hi @runjin326  thanks for adding this! 

Unfortunately, it's erroring out for me..

```
Using GeneSymbol as id variables
Error in normData[rownames(expressionMatrixMatched), ] : 
  subscript out of bounds
Calls: ZscoredAnnotation
```

I think it should be [[]] to get the items from the list `normData`
```suggestion
  normData = normData[[j]]
```
Just FYI, this is how the equivalent lapply run would look like:
```
GTExZscoredAnnotation_filtered_fusions_list <- lapply(
  c(1:length(cohortInterest)),
  function(j) {
  cohort_BSids <- read.delim(clinicalFile, header = TRUE, sep = "\t", stringsAsFactors = FALSE) %>%
    # uses each value x in cohortInterest
    filter(cohort == cohortInterest[j]) %>% filter(experimental_strategy == "RNA-Seq") %>%
    filter(sample_type == "Tumor") %>% pull("Kids_First_Biospecimen_ID")
  # use index to find the matched normal data
  normData = normData[[j]]
  print(dim(normData))
  GTExZscoredAnnotation_filtered_fusions_individual <-ZscoredAnnotation(standardFusionCalls,zscoreFilter,normData=normData,cohort_BSids=cohort_BSids, expressionMatrix = expressionMatrix)
  return(GTExZscoredAnnotation_filtered_fusions_individual)
  }
)  

GTExZscoredAnnotation_filtered_fusions <- do.call(rbind,GTExZscoredAnnotation_filtered_fusions_list ) 
saveRDS(GTExZscoredAnnotation_filtered_fusions,paste0(opt$outputfile,"_GTExComparison_annotated.RDS"))
```

Reply:
Sorry about that! My computer kept crashing when I run the entire code so I had to run it line by line and I didn't find that that error.